contrib/win32/win32compat/ssh-agent/agent.c

```diff
@@ -305,6 +305,8 @@ get_con_client_info(struct agent_connection* con)
 		GetTokenInformation(client_primary_token, TokenUser, info, info_len, &info_len) == FALSE)
 		goto done;
 
+	memset(info, 0, info_len);
```

Comment:
This doesn't seem right.  `info` is being zeroed out here, but then its fields, which will be zero, are used below.  I feel `info` should be zeroed out right after it is allocated.

`info->User.Sid`

Reply:
You are right, it seems that GetTokenInformation populates info (https://learn.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-gettokeninformation).
Maybe break that if statement into two if statements.